File: app/src/main/java/com/example/dreamloaf/viewmodels/AuthViewModel.java
package com.example.dreamloaf.viewmodels;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import com.example.dreamloaf.data.User;
import com.example.dreamloaf.repository.AuthRepository;
import java.util.concurrent.Executors;
public class AuthViewModel extends AndroidViewModel {
    private final AuthRepository authRepo;
    private final MutableLiveData<String> currentUserRole = new MutableLiveData<>();

    public AuthViewModel(@NonNull Application application) {
        super(application);
        authRepo = new AuthRepository(application);
    }

    public LiveData<Boolean> checkLoginAvailable(String login) {
        MutableLiveData<Boolean> result = new MutableLiveData<>();
        Executors.newSingleThreadExecutor().execute(() -> {
            result.postValue(!authRepo.isLoginExists(login));
        });
        return result;
    }

    public LiveData<Boolean> registerUser(User user) {
        MutableLiveData<Boolean> result = new MutableLiveData<>();
        Executors.newSingleThreadExecutor().execute(() -> {
            if (authRepo.isLoginExists(user.getLogin())) {
                result.postValue(false);
            } else {
                authRepo.register(user);
                currentUserRole.postValue(user.getRole());
                result.postValue(true);
            }
        });
        return result;
    }

    public LiveData<User> loginUser(String login, String password) {
        MutableLiveData<User> result = new MutableLiveData<>();
        Executors.newSingleThreadExecutor().execute(() -> {
            User user = authRepo.login(login, password);
            result.postValue(user);
        });
        return result;
    }
}

File: app/src/main/java/com/example/dreamloaf/viewmodels/AuthViewModel.kt
package com.example.dreamloaf.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.dreamloaf.data.User
import com.example.dreamloaf.repository.AuthRepository
import java.util.concurrent.Executors

class AuthViewModel(application: Application) : AndroidViewModel(application) {
    private val authRepo: AuthRepository
    private val currentUserRole: MutableLiveData<String?> = MutableLiveData<String?>()

    init {
        authRepo = AuthRepository(application)
    }

    fun checkLoginAvailable(login: String?): LiveData<Boolean?> {
        val result: MutableLiveData<Boolean?> = MutableLiveData<Boolean?>()
        Executors.newSingleThreadExecutor().execute(Runnable {
            result.postValue(!authRepo.isLoginExists(login))
        })
        return result
    }

    fun registerUser(user: User): LiveData<Boolean?> {
        val result: MutableLiveData<Boolean?> = MutableLiveData<Boolean?>()
        Executors.newSingleThreadExecutor().execute(Runnable {
            if (authRepo.isLoginExists(user.login)) {
                result.postValue(false)
            } else {
                authRepo.register(user)
                currentUserRole.postValue(user.role)
                result.postValue(true)
            }
        })
        return result
    }

    fun loginUser(login: String?, password: String?): LiveData<User?> {
        val result: MutableLiveData<User?> = MutableLiveData<User?>()
        Executors.newSingleThreadExecutor().execute(Runnable {
            val user = authRepo.login(login, password)
            result.postValue(user)
        })
        return result
    }
}
